apps/web/src/routes/AllView/AllView.jsx

```diff
@@ -94,15 +88,23 @@ export default function AllViews () {
     }))
   }, [updateContextWidget])
 
-  const widgetsSorted = useMemo(() => {
-    return visibleWidgets.map(widget => {
-      return ContextWidgetPresenter(widget, { t })
-    }).sort((a, b) => a.title.localeCompare(b.title))
-  }, [visibleWidgets])
+  // Filter and sort widgets and get them ready for display
+  const visibleWidgets = useMemo(() => {
+    return contextWidgets.filter(widget => {
+      // When not editing only show widgets with a related view or chat room
+      if (!isEditing && !widget.view && widget.type !== 'chat') return false
+      // When editing only show widgets that have not already been added
+      if (isEditing && widget.order) return false
```

Comment:
One trap that exists, something I have been meaning to bring up;

You can remove a widget that has children from the menu, and those children are still ordered (and thus cannot be accessed to add back to the menu themselves). It isn't obvious where they have ended up at this point but they would still be displayed in the All view. Now, with these changes, they will also be hidden.

Probably the best option is to dump all children from a widget if its removed from the menu (best done in both of the identical `reorderTree` functions).

Reply:
another good catch. yeah I think dumping them from the parent makes sense